scripts/portal-flow.js

```diff
@@ -1,7 +1,87 @@
-module.exports = function(runner) {
+'use strict';
+const configureRequest = require('../util/configureRequest');
+const requestBodyUtils = require('../util/sync_request_bodies');
+const recordUtils = require('../util/generate_record');
+const makeUser = require('../util/fixtures/makeUser');
+const makeWorkorder = require('../util/fixtures/makeWorkorder');
+const makeWorkflow = require('../util/fixtures/makeWorkflow');
+const makeMessage = require('../util/fixtures/makeMessage');
+const createUserAndGroup = require('../util/createUserAndGroup');
+const Promise = require('bluebird');
+
+function urlFor(baseUrl, dataset) {
+  return `${baseUrl}/mbaas/sync/${dataset}`;
+}
+
+function syncDataset(baseUrl, request, clientId, name) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    dataset_id: name,
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: []
+  });
+  return request.post({
+    url: urlFor(baseUrl, name),
+    body: payload,
+    json: true
+  }).then();
+}
+
+function createRecord(baseUrl, request, clientId, dataset, data) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    fn: 'sync',
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: [recordUtils.generateRecord(data)]
+  });
+  return request.post({
+    url: urlFor(baseUrl, 'workorders'),
```

Comment:
Nice catch! Refactoring brainfart...

On the validation side we've been thinking about adding runtime server-side schema validation, but I'm not sure how that would integrate with sync. Hopefully we'd be able to use the same validation client-side.
Also we might be able to rely a bit on supplying typescript-annotated utility functions from the modules to create the data, so have some compile-time checking there as well.